storage/innobase/srv/srv0srv.cc

```diff
@@ -1710,6 +1710,59 @@ void srv_update_purge_thread_count(uint n)
 
 Atomic_counter<int> srv_purge_thread_count_changed;
 
+/** get age of the redo log in lsn units
+@return percentage of redo log filled (calculated using lsn) */
+static lsn_t get_lsn_age_factor()
+{
+  mysql_mutex_lock(&buf_pool.flush_list_mutex);
+  lsn_t age = (log_sys.get_lsn() - buf_pool.get_oldest_modification(0));
+  mysql_mutex_unlock(&buf_pool.flush_list_mutex);
+  return ((age * 100) / log_sys.max_checkpoint_age);
+}
+
+/** distribute spread across N points.
+e.g.: spread of 60 is distrubuted across 4 points as: 6+12+18+24
+@param[in,out]	series	distributed spread series
+@param[in]	len	length of series
+@param[in]	spread	spread to distribute
+@param[in]	points	distribute spread across N points */
+void compute_series(ulint* series, const ulint len,
+	const ulint spread, const ulint points)
+{
+  memset(series, 0, sizeof(ulint) * len);
```

Comment:
This function should be declared `static`, and the static array `series` should be declared in the file scope (documented as being private to the thread that runs the purge coordinator), so that it does not need to be passed as a (constant) parameter.

We usually do not declare pass-by-value function parameters as `const`.

Would `uint8_t` suffice for the elements of the array?

Reply:
1. function made static.
2. moving series at global level with documentation? since the usage is limited to srv_do_purge() and is expected to get used as part of the logic wondering does it make more sense to declare and define it as part of the function.
3. "const" removed for pass-by-value though I thought it will clarify the usage.
4. uitn8_t can be replaced but I rarely see range-specific usage. defacto data-type for most of the variables I see is ulint but if you still recommend I can replace it.  like srv_n_purge_threads is ulint and likewise lot of other variables.